src/libopenrave/kinbodygeometry.cpp

```diff
@@ -563,19 +563,33 @@ inline void LoadJsonValue(const rapidjson::Value& v, KinBody::GeometryInfo::Side
     if(v.IsObject()) {
         orjson::LoadJsonValueByKey(v, "transform", t.transf);
         orjson::LoadJsonValueByKey(v, "halfExtents", t.vExtents);
-        std::string type = "";
-        orjson::LoadJsonValueByKey(v, "type", type);
-        std::map<std::string, KinBody::GeometryInfo::SideWallType> sideWallTypeMapping = {
-            {"nx", KinBody::GeometryInfo::SideWallType::SWT_NX},
-            {"px", KinBody::GeometryInfo::SideWallType::SWT_PX},
-            {"ny", KinBody::GeometryInfo::SideWallType::SWT_NY},
-            {"py", KinBody::GeometryInfo::SideWallType::SWT_PY},
-        };
-
-        if (sideWallTypeMapping.find(type) == sideWallTypeMapping.end()) {
-            throw OPENRAVE_EXCEPTION_FORMAT(_("unrecognized sidewall type %s for loading from json"), type, ORE_InvalidArguments);
+        if(!v.HasMember("type")) {
+            throw OPENRAVE_EXCEPTION_FORMAT0(_("need sidewall type when loading from json"), ORE_InvalidArguments);
+        }
+        int sideWallType = 0;
+        if (v["type"].IsInt()) {
+            // backward compatibility, support enum sideWall type
+            orjson::LoadJsonValueByKey(v, "type", sideWallType);
+            if (!(sideWallType >= KinBody::GeometryInfo::SideWallType::SWT_NX
+                && sideWallType <= KinBody::GeometryInfo::SideWallType::SWT_PY)) {
```

Comment:
Usually, I would do:

```c++
        enum SideWallType
        {
            SWT_NX=0,
            SWT_PX=1,
            SWT_NY=2,
            SWT_PY=3,
            SWT_First=SWT_NX,
            SWT_Last=SWT_PY,
        };
```

And test for `SWT_First` and `SWT_Last`, so that when someone change the header to add more types, they will not forget this place.